SoftLayer/managers/vs.py

```diff
@@ -874,9 +879,30 @@ def upgrade(self, instance_id, cpus=None, memory=None, nic_speed=None, public=Tr
                 raise exceptions.SoftLayerError(
                     "Unable to find %s option with value %s" % (option, value))
 
-            prices.append({'id': price_id})
-        order['prices'] = prices
-
+            if disk is not None:
+                category = {'categories': [{
+                    'categoryCode': 'guest_disk' + str(disk.get('number')),
+                    'complexType': "SoftLayer_Product_Item_Category"
+                }], 'complexType': 'SoftLayer_Product_Item_Price'}
+                prices.append(category)
+                prices[0]['id'] = price_id
+            elif add:
+                vsi_disk = self.get_instance(instance_id)
+                disk_number = 0
```

Comment:
I think you can just set this to `https://sldn.softlayer.com/reference/datatypes/SoftLayer_Virtual_Guest/#blockdevicecount` that way you can make a simple / quick API call instead of doing this nested loop.